### ISSUE TITLE
Classpath broken in shell with code-generating Cypher runtime

### ISSUE THREAD

**Issue by jotomo** [neo4j/neo4j#5122]
```
mal:~/inbox/neo4j-community-2.3.0-M02$ bin/neo4j-shell -path testdb
NOTE: Local Neo4j graph database service at 'testdb'
Welcome to the Neo4j Shell! Enter 'help' for a list of commands


neo4j-sh (?)$ create (n:Feed);
+-------------------+
| No data returned. |
+-------------------+
Nodes created: 1
Labels added: 1
1135 ms
neo4j-sh (?)$ match (n:Feed) return n;
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/lang3/StringEscapeUtils
    at org.neo4j.codegen.source.MethodWriter.constant(MethodWriter.java:267)
    at org.neo4j.codegen.Expression$7.accept(Expression.java:114)
    at org.neo4j.codegen.source.MethodWriter.arglist(MethodWriter.java:235)
    at org.neo4j.codegen.source.MethodWriter.invoke(MethodWriter.java:225)
    at org.neo4j.codegen.Expression$12.accept(Expression.java:177)
    at org.neo4j.codegen.source.ClassWriter.field(ClassWriter.java:137)
    at org.neo4j.codegen.ClassGenerator.emitField(ClassGenerator.java:97)
    at org.neo4j.codegen.ClassGenerator.staticField(ClassGenerator.java:82)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeStructure$$anon$1$$anonfun$1.apply(CodeStructure.scala:176)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeStructure$$anon$1$$anonfun$1.apply(CodeStructure.scala:163)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.package$.using(package.scala:34)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeStructure$$anon$1.generateQuery(CodeStructure.scala:163)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeStructure$$anon$1.generateQuery(CodeStructure.scala:155)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeGenerator$.generate(CodeGenerator.scala:53)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeGenerator$.generateQuery(CodeGenerator.scala:46)
    at org.neo4j.cypher.internal.compiler.v2_3.codegen.CodeGenerator.generate(CodeGenerator.scala:74)
    at org.neo4j.cypher.internal.compiler.v2_3.CompiledPlanBuilder$$anonfun$apply$2.apply(RuntimeBuilder.scala:117)
    at org.neo4j.cypher.internal.compiler.v2_3.CompiledPlanBuilder$$anonfun$apply$2.apply(RuntimeBuilder.scala:115)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.package$$anonfun$closing$1.apply(package.scala:26)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.package$$anonfun$closing$1.apply(package.scala:26)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.package$.using(package.scala:34)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.package$.closing(package.scala:26)
    at org.neo4j.cypher.internal.compiler.v2_3.CompiledPlanBuilder.apply(RuntimeBuilder.scala:115)
    at org.neo4j.cypher.internal.compiler.v2_3.RuntimeBuilder$class.apply(RuntimeBuilder.scala:49)
    at org.neo4j.cypher.internal.compiler.v2_3.SilentFallbackRuntimeBuilder.apply(RuntimeBuilder.scala:65)
    at org.neo4j.cypher.internal.compiler.v2_3.planner.CostBasedExecutablePlanBuilder.producePlan(CostBasedExecutablePlanBuilder.scala:69)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.FallbackBuilder$class.producePlan(SilentFallbackPlanBuilder.scala:42)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.SilentFallbackPlanBuilder.producePlan(SilentFallbackPlanBuilder.scala:65)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder.build(ExecutionPlanBuilder.scala:100)
    at org.neo4j.cypher.internal.compiler.v2_3.CypherCompiler$$anonfun$7$$anonfun$apply$1.apply(CypherCompiler.scala:173)
    at org.neo4j.cypher.internal.compiler.v2_3.CypherCompiler$$anonfun$7$$anonfun$apply$1.apply(CypherCompiler.scala:171)
    at org.neo4j.cypher.internal.compiler.v2_3.MonitoringCacheAccessor$$anonfun$1.apply(CacheAccessor.scala:33)
    at org.neo4j.cypher.internal.compiler.v2_3.LRUCache.getOrElseUpdate(LRUCache.scala:34)
    at org.neo4j.cypher.internal.compiler.v2_3.LRUCache.apply(LRUCache.scala:64)
    at org.neo4j.cypher.internal.compiler.v2_3.MonitoringCacheAccessor.getOrElseUpdate(CacheAccessor.scala:31)
    at org.neo4j.cypher.internal.compiler.v2_3.CypherCompiler$$anonfun$7.apply(CypherCompiler.scala:171)
    at org.neo4j.cypher.internal.compiler.v2_3.CypherCompiler$$anonfun$7.apply(CypherCompiler.scala:171)
    at scala.collection.Iterator$$anon$9.next(Iterator.scala:162)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at scala.collection.Iterator$$anon$12.next(Iterator.scala:397)
    at org.neo4j.cypher.internal.compiler.v2_3.CypherCompiler.planPreparedQuery(CypherCompiler.scala:182)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$$anon$2$$anonfun$plan$1.apply(CompatibilityFor2_3.scala:162)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$$anon$2$$anonfun$plan$1.apply(CompatibilityFor2_3.scala:160)
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.runSafely(CompatibilityFor2_3.scala:111)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$$anon$2.plan(CompatibilityFor2_3.scala:160)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3$$anonfun$liftedTree1$1$1$$anonfun$apply$1.apply(ExecutionEngine.scala:160)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3$$anonfun$liftedTree1$1$1$$anonfun$apply$1.apply(ExecutionEngine.scala:157)
    at org.neo4j.cypher.internal.compiler.v2_3.MonitoringCacheAccessor$$anonfun$1.apply(CacheAccessor.scala:33)
    at org.neo4j.cypher.internal.compiler.v2_3.LRUCache.getOrElseUpdate(LRUCache.scala:34)
    at org.neo4j.cypher.internal.compiler.v2_3.LRUCache.apply(LRUCache.scala:64)
    at org.neo4j.cypher.internal.compiler.v2_3.MonitoringCacheAccessor.getOrElseUpdate(CacheAccessor.scala:31)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3$$anonfun$liftedTree1$1$1.apply(ExecutionEngine.scala:157)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3$$anonfun$liftedTree1$1$1.apply(ExecutionEngine.scala:157)
    at scala.collection.Iterator$$anon$9.next(Iterator.scala:162)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at scala.collection.Iterator$$anon$12.next(Iterator.scala:397)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3.liftedTree1$1(ExecutionEngine.scala:169)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3.apply(ExecutionEngine.scala:149)
    at org.neo4j.cypher.ExecutionEngine$$anonfun$3.apply(ExecutionEngine.scala:134)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.package$.using(package.scala:34)
    at org.neo4j.cypher.ExecutionEngine.planQuery(ExecutionEngine.scala:134)
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:116)
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:111)
    at org.neo4j.cypher.javacompat.internal.ServerExecutionEngine.executeQuery(ServerExecutionEngine.java:62)
    at org.neo4j.shell.kernel.apps.cypher.Start.getResult(Start.java:87)
    at org.neo4j.shell.kernel.apps.cypher.Start.exec(Start.java:68)
    at org.neo4j.shell.kernel.apps.TransactionProvidingApp.execute(TransactionProvidingApp.java:223)
    at org.neo4j.shell.impl.AbstractAppServer.interpretLine(AbstractAppServer.java:117)
    at org.neo4j.shell.kernel.GraphDatabaseShellServer.interpretLine(GraphDatabaseShellServer.java:93)
    at org.neo4j.shell.impl.AbstractClient.evaluate(AbstractClient.java:149)
    at org.neo4j.shell.impl.AbstractClient.evaluate(AbstractClient.java:133)
    at org.neo4j.shell.impl.AbstractClient.grabPrompt(AbstractClient.java:101)
    at org.neo4j.shell.StartClient.grabPromptOrJustExecuteCommand(StartClient.java:375)
    at org.neo4j.shell.StartClient.tryStartLocalServerAndClient(StartClient.java:272)
    at org.neo4j.shell.StartClient.startLocal(StartClient.java:245)
    at org.neo4j.shell.StartClient.start(StartClient.java:178)
    at org.neo4j.shell.StartClient.main(StartClient.java:134)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.StringEscapeUtils
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 77 more
```

**Comment by jotomo** [comment]
Only applies when directly opening a db, rather than connecting to a server, and thus running Cypher locally.

**Comment by lutovich** [comment]
Hello @jotomo,

I think it was fixed by https://github.com/neo4j/neo4j/pull/4794 and just not released yet.
Fix will be available with next 2.3.x.

**Comment by jotomo** [comment]
Local snapshot works indeed. Thanks @lutovich
